Adding square-root exercise

evelynstender opened the pull request:
Closes #906

SleeplessByte commented on exercises/square-root/example.js:
```diff
@@ -0,0 +1,3 @@
+export const squareRoot = (radicand = 0) => {
+  return Math.sqrt(radicand);
```
I think the point is to _not_ use a built-in function 😅

SleeplessByte reviewed (commented):
I think we should clarify we don't want to use `Math.sqrt`. We might check other languages that have implemented this exercise to confirm that other tracks also disallow using built-in functions.

If this is the case:

- add `.meta/hints.md` and disallow `Math.sqrt`
- `bin/configlet generate . --only square-root` again to update the README.md
- update the example solution

evelynstender commented:
Oh yeah! That makes sense! I didn't realise we couldn't use `Math.sqrt()`. I didn't see it anywhere haha

I will update! Thanks!

evelynstender commented on config.json:
```diff
@@ -227,6 +227,14 @@
+      "slug": "square-root",
+      "uuid": "31c7725e-def0-4dc9-88a0-e19afb9bfb06",
+      "core": false,
+      "unlocked_by": null,
```
I'm not sure which exercise should unlock this (if any)

evelynstender commented:
@SleeplessByte I updated the example, added the `hint.md` and updated the readme file. I'm not sure which exercise should unlock this (if any), so I left it as `null`

SleeplessByte reviewed (approved):
Perfect!

SleeplessByte commented:
@evelynstender in v2, _only_ core exercises can unlock anything, and since this is a reimplementation and mathy exercise, it will never become core.
